cg/cli/backup.py

```diff
@@ -1,32 +1,110 @@
 """Backup related CLI commands."""
 import logging
 from pathlib import Path
-from typing import Iterable, List, Optional, Union
+from typing import Iterable, List, Optional, Tuple, Union
 
 import click
 import housekeeper.store.models as hk_models
 
 from cg.apps.housekeeper.hk import HousekeeperAPI
+from cg.constants import FileExtensions
 from cg.constants.constants import DRY_RUN, FlowCellStatus
 from cg.constants.housekeeper_tags import SequencingFileTag
+from cg.exc import FlowCellError
 from cg.meta.backup.backup import BackupAPI, SpringBackupAPI
 from cg.meta.backup.pdc import PdcAPI
-from cg.meta.encryption.encryption import EncryptionAPI, SpringEncryptionAPI
+from cg.meta.encryption.encryption import (
+    EncryptionAPI,
+    FlowCellEncryptionAPI,
+    SpringEncryptionAPI,
+)
 from cg.meta.tar.tar import TarAPI
 from cg.models.cg_config import CGConfig
+from cg.models.demultiplex.flow_cell import FlowCellDirectoryData
 from cg.store import Store
 from cg.store.models import Flowcell, Sample
 
 LOG = logging.getLogger(__name__)
 
 
+def _is_encryption_possible(
+    db_flow_cell: Flowcell,
+    flow_cell: FlowCellDirectoryData,
+    complete_file_path: Path,
+    pending_file_path: Path,
+) -> Tuple[bool, Union[str, None]]:
+    """Check if requirements for encryption are meet."""
+    if db_flow_cell and db_flow_cell.has_backup:
+        return False, f"Flow cell: {flow_cell.id} is already backed-up"
+    if not flow_cell.is_flow_cell_ready():
+        return False, f"Flow cell: {flow_cell.id} is not ready"
+    if complete_file_path.exists():
+        return False, f"Encryption already completed for flow cell: {flow_cell.id}"
+    if pending_file_path.exists():
+        return False, f"Encryption already started for flow cell: {flow_cell.id}"
+    return True, None
+
+
 @click.group()
 @click.pass_obj
 def backup(context: CGConfig):
     """Backup utilities"""
     pass
 
 
+@backup.command("encrypt-flow-cells")
+@DRY_RUN
+@click.pass_obj
+def encrypt_flow_cells(context: CGConfig, dry_run: bool):
```

Comment:
I think it is good if we try to keep as little logic as possible in the cli commands. Maybe some of the logic here could be moved into the encryption api?

Reply:
It could look something like
```python
@backup.command("encrypt-flow-cells")
@DRY_RUN
@click.pass_obj
def encrypt_flow_cells(context: CGConfig, dry_run: bool):
    """Encrypt flow cells."""
    status_db: Store = context.status_db
    flow_cell_encryption_api = FlowCellEncryptionAPI(
        binary_path=context.encryption.binary_path, config=context.dict(), dry_run=dry_run
    )
    flow_cell_encryption_api.slurm_api.set_dry_run(dry_run)
    encrypt_dir: Path = Path(context.backup.encrypt_dir)
    flow_cells_dir = Path(context.flow_cells_dir)
    
    flow_cell_encryption_api.initiate_encryption(flow_cells_dir, encrypt_dir, status_db)
```
Also, I think we generally pass the api via the context? So if we want to stick to that pattern, maybe `FlowCellEncryptionAPI` should be passed in the context like `status_db` is

Reply:
We have to instantiate per flow cell so by context is not possible.